java-components/cache/src/main/java/com/redhat/hacbs/artifactcache/artifactwatch/ScmLookup.java

```diff
@@ -83,9 +83,17 @@ public void onAdd(ArtifactBuild newObj) {
                             newObj.getStatus().setMessage("");
                             newObj.getStatus().setState(ArtifactBuild.DISCOVERING);
                         } catch (Exception e) {
-                            Log.errorf(e, "Failed to update rebuilt object");
+                            Log.errorf(e, "Failed to update rebuilt object " + i);
                             newObj.getStatus().setMessage(e.getMessage());
                             newObj.getStatus().setState(ArtifactBuild.MISSING);
+                            // Matches status labels in pkg/reconciler/util/util.go
```

Comment:
Don't update the labels here. The operator will take care of it. If you do an update here the operator will attempt to reverse the change, as the status has not been updated yet, and will likely cause 'already modified' exceptions.

This whole approach of doing this change in the cache is not 100% idiomatic, but saves a huge amount of memory and CPU. Basically ideally the operator would kick of a pipeline, the pipeline would return this info in a result, and then the operator would act on it to update the objects, but that kicks off hundreds of pipelines in some cases.